Fix wallmount board dropping at random side of the wall

qwerltaz opened the pull request:
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
<!-- What did you change in this PR? -->
During deconstruction, many devices drop the board at their own coordinates. This is mostly fine, but when the device is wall-mounted, the board will warp to a random side of the wall, sometimes opposite of the player deconstructing it, and sometimes inside the wall. I change this so the board is picked up when the player has free hands, otherwise, it drops under the player's feet.

## Technical details
<!-- If this is a code change, summarize at high level how your new code works. This makes it easier to review. -->
Add `EmptyAtUser` data field to `EmptyAllContainers`. When true, get user coords and empty the container there. 

## Media
<!-- 
PRs which make ingame changes (adding clothing, items, new features, etc) are required to have media attached that showcase the changes.
Small fixes/refactors are exempt.
Any media may be used in SS14 progress reports, with clear credit given.

If you're unsure whether your PR will require media, ask a maintainer.

Check the box below to confirm that you have in fact seen this (put an X in the brackets, like [X]):
-->
The boards drop at my feet, instead of under the wall, inside it, or on the other side.
![boards working](https://github.com/space-wizards/space-station-14/assets/69696513/1e65ae71-fc9d-47f5-b094-cc1b83df1c6e)

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
<!--
Make players aware of new features and changes that could affect how they play the game by adding a Changelog entry. Please read the Changelog guidelines located at: https://docs.spacestation14.io/en/getting-started/pr-guideline#changelog
-->

metalgearsloth commented on Content.Server/Construction/Completions/EmptyAllContainers.cs:
```diff
@@ -28,9 +36,13 @@ public void PerformAction(EntityUid uid, EntityUid? userUid, IEntityManager enti
             HandsComponent? hands = null;
             var pickup = Pickup && entityManager.TryGetComponent(userUid, out hands);
 
+            EntityCoordinates? dropCoordinates = null;
+            if (EmptyAtUser && entityManager.TryGetComponent(userUid, out TransformComponent? userXform))
+                dropCoordinates = userXform.Coordinates;
+
             foreach (var container in containerManager.GetAllContainers())
             {
-                foreach (var ent in containerSys.EmptyContainer(container, true, reparent: !pickup))
+                foreach (var ent in containerSys.EmptyContainer(container, true, dropCoordinates, reparent: !pickup))
                 {
```
Does this work properly if the user is parented? E.g. buckled, as I feel this is just going to actually use the coordinates which isn't valid behavior.

qwerltaz commented:
- Changed to DropNextTo
- Added doors to the family as they become solid when deconstructing, leading to similar behavior as with wallmounts.

(The code refactor in [2191c44](https://github.com/space-wizards/space-station-14/pull/21920/commits/2191c44c3843dc63744d86788e911b27551d0f12) was unintentional.)
